authLdap.php

```diff
@@ -263,39 +262,40 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
         }
 
         $uid = authLdap_get_uid($realuid);
-        $role = '';
-
-        // we only need this if either LDAP groups are disabled or
-        // if the WordPress role of the user overrides LDAP groups
-        if (!$authLDAPGroupEnable || !$authLDAPGroupOverUser) {
-            $role = authLdap_user_role($uid);
-        }
+        $roles = authLdap_user_roles($uid);
```

Comment:
All this does is get the current role(s).  The point is to not clobber existing roles but to gracefully merge additional ones (unless configured to completely override).  This line simply gets the roles as they exist and then adds them as roles at the end of the function.  So if LDAP is not configured to override then these are simply added back in and no harm done.  What am I missing?